### PR TITLE
Create apicspinecollector.py

### PR DESCRIPTION
**Pull request by rahul61185**
@IvoGoman @artherd42 Please review this PR for spine port availability tracker

### CODE REVIEW

**Comment by rahul61185** [comment]
@artherd42 @IvoGoman Could you please review this?

**Comment by rahul61185** [comment]
@IvoGoman I have used vs code to create the script but looks like while copying it in browser, indentation moved.
I have also adjusted the script based on your suggestions. Could you please review it again.

File: collectors/apicspineportcollector.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+          'used_port_count',
+            'Total in-use ports',
+            labels=['apicHost', 'Spine_id', 'podId'])
+
+        g_down_port = GaugeMetricFamily(
+            'down_port_count',
+            'In-use but down ports',
+            labels=['apicHost', 'Spine_id', 'podId'])
+
+        metric_counter = 0
```
Comment (artherd42) [inline collectors/apicspineportcollector.py:42]: can be removed. already defined in #12

File: collectors/apicspineportcollector.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                    # Used Ports
+                    g_used_port.add_metric(
+                        labels=[host, spine_id, 'podId'],
+                        value=used_port_count
+                    
+                    # Down ports
+                    g_down_port.add_metric(
+                        labels=[host, spine_id, 'podId'],
+                        value=down_port_count
+            break  # Each host produces the same metrics.
```
Comment (artherd42) [inline collectors/apicspineportcollector.py:88]: **add self.__metric_counter += 1** before the break

File: collectors/apicspineportcollector.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+        yield g_free_port_count
+        yield g_used_port_count
+        yield g_down_port_count
+
+        LOG.info('Collected %s APIC Spine ports metrics')
```
Comment (artherd42) [inline collectors/apicspineportcollector.py:94]: should be **LOG.info('Collected %s APIC Spine ports metrics', self.__metric_counter)**

File: collectors/apicspineportcollector.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+                    # Used Ports
+                    g_used_port.add_metric(
+                        labels=[host, spine_id, 'podId'],
+                        value=used_port_count
+                    
+                    # Down ports
+                    g_down_port.add_metric(
+                        labels=[host, spine_id, 'podId'],
+                        value=down_port_count
```
Comment (artherd42) [inline collectors/apicspineportcollector.py:87]: add self.__metric_counter += 1

File: collectors/apicspineportcollector.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+                            free_port_count += 1
+                        elif (port_dict['l1PhysIf']['attributes']['adminSt'] == 'up') and (port_dict['l1PhysIf']['children'][0]['ethpmPhysIf']['attributes']['operSt'] == 'up'):
+                            used_port_count = port_dict['l1PhysIf']['attributes']['id']
+                            used_port_count +=1
+                        elif port_dict['l1PhysIf']['attributes']['adminSt'] == 'down':
+                            down_port_count = port_dict['l1PhysIf']['attributes']['id']
+                            down_port_count +=1
+                    # Free Ports
+                    g_free_port.add_metric(
+                        labels=[host, spine_id, 'podId'],
```
Comment (artherd42) [inline collectors/apicspineportcollector.py:76]: This sets the value of the label podId to the string podId. Guess you want to have the actual pod_id as the label value. Then labels should be **labels=[host, spine_id, pod_id]**

**Comment by rahul61185** [comment]
@artherd42  Could you please review it again?

**Review by artherd42 — APPROVED** [review]
lgtm

**Comment by artherd42** [comment]
@rahul61185 Running the collector fails with error below. Did you test your code locally?

```
Exception happened during processing of request from ('127.0.0.1', 54132)
Traceback (most recent call last):
  File "/Users/d021500/Developer/tools/miniconda2/envs/infrastructure-exporters/lib/python3.6/socketserver.py", line 654, in process_request_thread
    self.finish_request(request, client_address)
  File "/Users/d021500/Developer/tools/miniconda2/envs/infrastructure-exporters/lib/python3.6/socketserver.py", line 364, in finish_request
    self.RequestHandlerClass(request, client_address, self)
  File "/Users/d021500/Developer/tools/miniconda2/envs/infrastructure-exporters/lib/python3.6/socketserver.py", line 724, in __init__
    self.handle()
  File "/Users/d021500/Developer/tools/miniconda2/envs/infrastructure-exporters/lib/python3.6/http/server.py", line 418, in handle
    self.handle_one_request()
  File "/Users/d021500/Developer/tools/miniconda2/envs/infrastructure-exporters/lib/python3.6/http/server.py", line 406, in handle_one_request
    method()
  File "/Users/d021500/Developer/tools/miniconda2/envs/infrastructure-exporters/lib/python3.6/site-packages/prometheus_client/exposition.py", line 146, in do_GET
    output = encoder(registry)
  File "/Users/d021500/Developer/tools/miniconda2/envs/infrastructure-exporters/lib/python3.6/site-packages/prometheus_client/exposition.py", line 89, in generate_latest
    for metric in registry.collect():
  File "/Users/d021500/Developer/tools/miniconda2/envs/infrastructure-exporters/lib/python3.6/site-packages/prometheus_client/registry.py", line 75, in collect
    for metric in collector.collect():
  File "/Users/d021500/Developer/git/apic-exporter/collectors/apicspineport.py", line 67, in collect
    free_port_count += 1
TypeError: must be str, not int
```
----------------------------------------

**Pull request merged**